### ISSUE TITLE
Texture Sharing does not install properly on Blender 4.1 under macOS Sonoma 14.5

### ISSUE THREAD

**Issue by christophepersoz** [maybites/blender.script.spout#33]
Hello,

I tried to follow the steps and do not forgot anything. pip is updated to its latests version, but I couldn't get Texture Sharing add-on installed. I'm getting the following report when I try to activate it,

```
Traceback (most recent call last):
File "/Applications/Blender.app/Contents/Resources/4.1/scripts/modules/addon_utils.py", line 376, in enable
mod = importlib.import_module(module_name)
File "/Applications/Blender.app/Contents/Resources/4.1/python/lib/python3.11/importlib/_init_py", line 126, in import_module return _bootstrap._gcd import (name (level:), package, level)
File "< frozen importlib._bootstrap>", line 1204, in _gcd_import
File "< frozen importlib._bootstrap>", line 1176, in _find_and_load
File "< frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
File "< frozen importlib._bootstrap>", line 241, in _call with_frames_removed
File "< frozen importlib._bootstrap>", line 1204, in _gcd_import
File "< frozen importlib._bootstrap>", line 1176, in _find_and_load
File "< frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
File "< frozen importlib._bootstrap>", line 241, in _call with_frames_removed
File "< frozen importlib._bootstrap>", line 1204, in _gcd_import
File "< frozen importlib._bootstrap>", line 1176, in _find_and_load
File "< frozen importlib._bootstrap>", line 1140, in _find_and_load _unlocked
ModuleNotFoundError: No module named 'blender-texture-sharing-6"
```
![Screenshot 2024-06-20 at 21 45 00](https://github.com/maybites/blender-texture-sharing/assets/12657220/679f0d97-318b-4eda-b36d-755ac83a8393)

I did install from terminal manually the syphon-python library with following command,
`pip install --upgrade syphon-python==0.1.0`
which returns,
```DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Collecting syphon-python==0.1.0
Downloading syphon_python-0.1.0-cp39-cp39-macosx_10_9_universal2.whl.metadata (2.3 kB)
Collecting pyobjc-core~=10.1 (from syphon-python==0.1.0)
Downloading pyobjc_core-10.3.1-cp39-cp39-macosx_10_9_universal2.whl.metadata (2.5 kB)
Collecting pyobjc-framework-metal~=10.1 (from syphon-python==0.1.0)
Downloading pyobjc_framework_Metal-10.3.1-cp36-abi3-macosx_10_13_universal2.whl.metadata (2.2 kB)
Collecting pyobjc-framework-cocoa~=10.1 (from syphon-python==0.1.0)
Downloading pyobjc_framework_Cocoa-10.3.1-cp39-cp39-macosx_10_9_universal2.whl.metadata (2.3 kB)
Collecting pyopengl~=3.1.7 (from syphon-python==0.1.0)

Downloading PyOpenGL-3.1.7-py3-none-any.whl.metadata (3.2 kB)
Downloading syphon_python-0.1.0-cp39-cp39-macosx_10_9_universal2.whl (458 kB)
    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 458.8/458.8 kB 6.4 MB/s eta 0:00:00
Downloading pyobjc_core-10.3.1-cp39-cp39-macosx_10_9_universal2.whl (780 kB)
    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 780.3/780.3 kB 23.8 MB/s eta 0:00:00
Downloading pyobjc_framework_Cocoa-10.3.1-cp39-cp39-macosx_10_9_universal2.whl (396 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 396.1/396.1 kB 29.3 MB/s eta 0:00:00
Downloading pyobjc_framework_Metal-10.3.1-cp36-abi3-macosx_10_13_universal2.whl (55 kB)
    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.7/55.7 kB 5.2 MB/s eta 0:00:00
Downloading PyOpenGL-3.1.7-py3-none-any.whl (2.4 MB)
    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.4/2.4 MB 50.3 MB/s eta 0:00:00
Installing collected packages: pyopengl, pyobjc-core, pyobjc-framework-cocoa, pyobjc-framework-metal, syphon-python
    DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621

Successfully installed pyobjc-core-10.3.1 pyobjc-framework-cocoa-10.3.1 pyobjc-framework-metal-10.3.1 pyopengl-3.1.7 syphon-python-0.1.0
```

I also install manually the NDI python with the following command, `pip install --upgrade ndi-python==5.1.1.1`
```
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Collecting ndi-python==5.1.1.1
  Downloading ndi_python-5.1.1.1-cp39-cp39-macosx_10_12_x86_64.whl.metadata (1.2 kB)
Requirement already satisfied: numpy in /usr/local/lib/python3.9/site-packages (from ndi-python==5.1.1.1) (1.21.1)
Downloading ndi_python-5.1.1.1-cp39-cp39-macosx_10_12_x86_64.whl (4.6 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.6/4.6 MB 5.4 MB/s eta 0:00:00
Installing collected packages: ndi-python
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Successfully installed ndi-python-5.1.1.1
```

Aside Deprecation warning, the installs seems correct but even with all this done I couldn't get it worked.

Is there anything to do to get that working here?
Thanks a lot,

Christophe

**Comment by maybites** [comment]
`ModuleNotFoundError: No module named 'blender-texture-sharing-6"`

I am curious about this line.

how did you install the addon?

**Comment by christophepersoz** [comment]
Quite simply, I downloaded the zip file from your repo and inside Blender > Preference > Install choose the `blender-texture-sharing-6.0.5.zip` file
Then, check the box and got the screenshot I posted.

**Comment by maybites** [comment]
which zip file? please post the link.

**Comment by christophepersoz** [comment]
> which zip file? please post the link.

This one, https://github.com/maybites/blender-texture-sharing/releases/download/V6.0.5/blender-texture-sharing.zip

**Comment by maybites** [comment]
`ModuleNotFoundError: No module named 'blender-texture-sharing-6"`

do you have any explanation why there is this error?

this points to a naming issue of the addon..

**Comment by christophepersoz** [comment]
> `ModuleNotFoundError: No module named 'blender-texture-sharing-6"`
> 
> do you have any explanation why there is this error?
> 
> this points to a naming issue of the addon..

Unfortunately no, I have no explanation, that's the whole point and why I post the issue there. 
I'm going to uninstall what I did yesterday, and restart the whole procedure from the beginning and see if things will ends correctly or not.
I will keep you in touch.

Thanks.

**Comment by christophepersoz** [comment]
I could finally install the package, I just do not know why it messed up yesterday. I just remove it and re-install from the zip file like I did on the first time yesterday. So, problem solved.

PS: Can Blender input a NDI texture onto Viewport Background or Object?
I tried to get it but showed nothing in Blender interface. I used TouchDesigner NDI Out to get the stream, and check that stream with NDI tools and it works. In blender I could get the stream in UV Mapping interface, but never saw any stream in Blender. Is that possible? Is there a trick or something special to do to get that work? Thanks.
